components/dash-core-components/src/components/Tooltip.react.js

```diff
@@ -220,7 +367,13 @@ Tooltip.propTypes = {
     /**
      * The side of the `bbox` on which the tooltip should open.
      */
-    direction: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
+    direction: PropTypes.oneOf(['top', 'right', 'bottom', 'left', 'auto']),
+
+    /**
+     * Query selector for container in which tooltip should be visible.
+     * If not set the tooltip`s parent element will be used.
+     */
+    auto_direction_container: PropTypes.string,
```

Comment:
I imagine a common ask would be just to make sure the tooltip doesn't go off the screen viewport (which typically won't change mid-hover) - is there a way we could allow that?

Reply:
In fact I guess there are three flavors you might want: just keep it on screen, just keep it in the specified container, or keep it in the intersection of the screen and the specified container.

Not sure what API would be best for this... maybe in addition to query selectors, `auto_direction_container` could accept either `'screen'` or `'screen,<selector>'` - the latter kind of looks like a selector for both "screen" and your container.

Also maybe `'window'` is better than `'screen'`?